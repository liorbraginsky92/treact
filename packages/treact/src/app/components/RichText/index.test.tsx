import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { RichText } from './'

it('RichText renders correctly without data', () => {
  const tree = renderer.create(
    <RichText
      id={1}
      entities={[]}
      text="" />,
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

it('RichText renders correctly with data', () => {
  const msg = {
    entities: [
      {
        offset: 14,
        length: 6,
        _: 'messageEntityCode',
      },
      {
        length: 5,
        offset: 25,
        _: 'messageEntityCode',
      },
      {
        length: 11,
        offset: 48,
        _: 'messageEntityUrl',
      },
      {
        length: 6,
        offset: 60,
        _: 'messageEntityUrl',
      },
    ],
    message: 'abc123 abc123 abc123 **___*_*`_** 😃😂😏😁🅱❤👅 youtube.com goo.gl must show',
  }

  const tree = renderer.create(
    <RichText
      id={2}
      entities={msg.entities}
      text={msg.message} />,
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
