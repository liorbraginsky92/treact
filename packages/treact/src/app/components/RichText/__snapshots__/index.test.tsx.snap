// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText renders correctly with data 1`] = `
<span>
  <span>
    abc123 abc123 
  </span>
  <code>
    abc123
  </code>
  <span>
     **__
  </span>
  <code>
    _*_*\`
  </code>
  <span>
    _** 😃😂😏😁🅱❤👅 
  </span>
  <a
    href="youtube.com"
  >
    youtube.com
  </a>
  <span>
     
  </span>
  <a
    href="goo.gl"
  >
    goo.gl
  </a>
  <span>
     must show
  </span>
</span>
`;

exports[`RichText renders correctly without data 1`] = `
<span>
  <span>
    
  </span>
</span>
`;
